### PR TITLE
GameCube Config: Store paths relatively when selected file is within Dolphin's directory. (Windows)

### PR DESCRIPTION
**Pull request by AdmiralCurtiss**
This is for portable installations that have the User dir in the same directory as Dolphin. Currently, if you pick a Memory Card file in that setup, then move the Dolphin dir somewhere else, Dolphin will no longer find the Memory Card file and force you to re-select it. There seems to have been some code that handled such cases already, but it relied on `GetExeDirectory()` returning a path that ends in a directory separator, which is not (or no longer?) the case. It also probably wouldn't have worked anyway as it incremented the returned character by one before comparing it to a directory separator -- I assume the `+ 1` was meant to go inside of the `at()` call?

Anyway, this fixes that without affecting regular installs that have their User dir in My Documents or whatever.

I've also fixed a small issue where `GetExeDirectory()` would return non-normalized paths, mainly because the way Visual Studio launches executables results in a path that includes `dolphin\Source\..\Binary` being returned by `GetModuleFileName()`, which then interfered with the detection of the "is this file in Dolphin's dir?" part of the memory card path set function.

Fixes [issue 8137](https://code.google.com/p/dolphin-emu/issues/detail?id=8137) for Windows only. I don't know how the User dir and portable install situation looks on other OSes.

### CODE REVIEW

**Comment by AdmiralCurtiss** [comment]
I just dug back and found 8c103a8deea777c3dbb65b805407846f5fe1252e, which I probably actually break again with this. I guess I misinterpreted what that code was supposed to do. Lemme re-fix that...

**Comment by AdmiralCurtiss** [comment]
There we go, much simpler too.

File: Source/Core/DolphinWX/Config/GameCubeConfigPane.cpp [hunk 1]
```diff
@@ -332,22 +333,21 @@ void GameCubeConfigPane::ChooseSlotPath(bool is_slot_a, TEXIDevices device_type)
...
-			if (next_char == '/' || next_char == '\\')
-			{
-				filename.erase(0, File::GetExeDirectory().size() + 1);
-				filename = "./" + filename;
-			}
-		}
+		// If the Memory Card file is within the Exe dir, we can assume that the user wants it to be stored relative
+		// to the executable, so it stays set correctly when the probably portable Exe dir is moved.
+		std::string exeDir = File::GetExeDirectory() + '\\';
+		if (!strncmp(exeDir.c_str(), filename.c_str(), exeDir.size()))
```
Comment (BhaaLseN) [inline Source/Core/DolphinWX/Config/GameCubeConfigPane.cpp]: Not sure, but isn't there a C++/STL equivalent for that? Also, maybe consider doing a case-insensitive comparison?

Comment (AdmiralCurtiss) [inline Source/Core/DolphinWX/Config/GameCubeConfigPane.cpp]: Yeah good point, I'll look into that.

Comment (AdmiralCurtiss) [inline Source/Core/DolphinWX/Config/GameCubeConfigPane.cpp]: So it seems like the closest thing is [std::experimental::filesystem](http://en.cppreference.com/w/cpp/experimental/fs), which is unfortunately not part of the C++ standard yet. I think the best thing we can do for now is the case-insensitive comparison of the normalized absolute paths.

**Comment by AdmiralCurtiss** [comment]
Would this be better, maybe? If yes I'll squash.

**Comment by BhaaLseN** [comment]
Usually, when dealing with localization stuff (think of the username inside the path), just calling lower/upper on the sides might not be the correct solution (google for the Turkish I Problem if you want to know more). But on the other hand, its probably as good as any other solution, so I don't really care to use one over the other at this point (until a better solution is indeed available).

**Pull request merged**